index.js

```diff
@@ -279,7 +279,7 @@ function $asBooleanNullable (bool) {
 function $asDatetime (date) {
   if (date instanceof Date) {
     return '"' + date.toISOString() + '"'
-  } else if (typeof date.toISOString === 'function') {
+  } else if (date && typeof date.toISOString === 'function') {
```

Comment:
could you add this also in $asTime?

Reply:
sure would like to know your thoughts on this

```javascript
function $asDatetime (date) {
  if (!date) {
    return '""'
  }

  if (date instanceof Date) {
    return '"' + date.toISOString() + '"'
  } else if (typeof date.toISOString === 'function') {
    return '"' + date.toISOString() + '"'
  } else {
    return $asString(date)
  }
}
```

for both `$asDateTime` & `$asTime` makes sense?

Reply:
My only concern is that we are not using the `$asString` function in this way, and we are managing a string in the date function format, for this I would put this check in the `else if` as it was before

Reply:
we also need it in `$asDate`?

```javascript
function $asDate (date) {
  if (date instanceof Date) {
    var year = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(date)
    var month = new Intl.DateTimeFormat('en', { month: '2-digit' }).format(date)
    var day = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(date)
    return '"' + year + '-' + month + '-' + day + '"'
  } else if (date && typeof date.format === 'function') {
    return '"' + date.format('YYYY-MM-DD') + '"'
  } else {
    return $asString(date)
  }
}
```

Reply:
yes, it is! ✨